src/Microsoft.Azure.NotificationHubs/NamespaceManager.cs

```diff
@@ -466,8 +466,31 @@ public Task<NotificationHubDescription> UpdateNotificationHubAsync(NotificationH
             throw new NotImplementedException();
         }
 
-        private static string AddHeaderAndFooterToXml(string content) => $"{Header}{content}{Footer}";
+        /// <summary>Checks whether a notifications hub exists.</summary>
+        /// <param name="path">The notification hub path.</param>
+        /// <returns>True if the hub exists</returns>
+        public bool NotificationHubExists(string path) => 
+            NotificationHubExistsAsync(path).GetAwaiter().GetResult();
 
+        /// <summary>
+        /// Checks whether a notification hub exists asynchronously.
+        /// </summary>
+        /// <param name="path">The notification hub path.</param>
+        /// <returns>A task that represents the asynchronous hub check operation</returns>
+        public async Task<bool> NotificationHubExistsAsync(string path)
+        {
+            try
+            {
+                var hubDescription = await GetNotificationHubAsync(path);
+                return hubDescription.Path == path;
```

Comment:
New to the team, forgive what may be a naive question:
Should this check be case insensitive? If it's flexible on the server, I'd really like it to be flexible here too, just as a defensive measure.

Reply:
Nice catch! Fixed